### ISSUE TITLE
Installation issue: mumps (serial), error "/bin/sh: line 0: fc: -h: invalid option"

### ISSUE THREAD

**Issue by samfux84** [spack/spack#11498]
### Steps to reproduce the issue

`spack install mumps@5.1.1+scotch~ptscotch~mpi+metis~parmetis+float+double+complex ^scotch@6.0.3+esmumps~mpi %gcc@4.8.5 arch=linux-centos7-x86_64`

Error:

```
link_cmd() { fc -shared -Wl,-soname -Wl,/cluster/spack/apps/linux-centos7-x86_64/gcc-4.8.5/mumps-5.1.1-vvioxgwnqlr7rrdpzoktm3batzppozbi/lib/libpord.so -o "$@" -L/cluster/spack/apps/linux-centos7-x86_64/gcc-4.8.5/scotch-6.0.3-zp7hpy7mezr5betm6bdbddmifa3oim4d/lib -L/cluster/spack/apps/linux-centos7-x86_64/gcc-4.8.5/zlib-1.2.11-tu4n3a4rud6dtfqmgmgfrb32liysc5lu/lib -lesmumps -lscotch -lscotcherr -lz -L/cluster/spack/apps/linux-centos7-x86_64/gcc-4.8.5/metis-5.1.0-iennmlkbyexvyt5labjpnf6jwh7yo4tr/lib -lmetis -L/cluster/spack/apps/linux-centos7-x86_64/gcc-4.8.5/openblas-0.2.20-ut6yhgxs5opzdjsqbfm7kaodfpqj7haf/lib -lopenblas; }; link_cmd libpord.so graph.o gbipart.o gbisect.o ddcreate.o ddbisect.o nestdiss.o multisector.o gelim.o bucket.o tree.o symbfac.o interface.o sort.o minpriority.o
/bin/sh: line 0: fc: -h: invalid option
fc: usage: fc [-e ename] [-lnr] [first] [last] or fc -s [pat=rep] [command]
make[2]: *** [Makefile:29: libpord.so] Error 2
make[2]: Leaving directory '/dev/shm/spackapps/spack-stage/spack-stage-Q5M9wr/MUMPS_5.1.1/PORD/lib'
make[1]: *** [Makefile:66: lib/libpord.so] Error 2
make[1]: *** Waiting for unfinished jobs....
```

### Platform and user environment

Please report your OS here:
```
[spackapps@eu-c7-042-05 scripts]$ uname -a
Linux eu-c7-042-05 3.10.0-862.14.4.el7.x86_64 #1 SMP Wed Sep 26 15:12:11 UTC 2018 x86_64 GNU/Linux
```

```
[spackapps@eu-c7-042-05 scripts]$ lsb_release -d
Description:    CentOS Linux release 7.5.1804 (Core)
```

```
- compiler:
    environment: {}
    extra_rpaths: []
    flags:
      cflags: -O2 -fPIC -ftree-vectorize -march=core-avx2 -mavx2
      cppflags: -O2 -fPIC -ftree-vectorize -march=core-avx2 -mavx2
      cxxflags: -O2 -fPIC -ftree-vectorize -march=core-avx2 -mavx2
    modules: []
    operating_system: centos7
    paths:
      cc: /usr/bin/gcc
      cxx: /usr/bin/g++
      f77: /usr/bin/gfortran
      fc: /usr/bin/gfortran
    spec: gcc@4.8.5
    target: x86_64
```

### Additional information

Building the same spec with +mpi instead of ~mpi worked without problems.

The main difference of the mumps Makefile.inc with respect to FC or fc, when comparing +mpi with ~mpi is:

```
        if '+mpi' in self.spec:
            scalapack = self.spec['scalapack'].libs if not shared \
                else LibraryList([])
            makefile_conf.extend(
                ['CC = {0}'.format(self.spec['mpi'].mpicc),
                 'FC = {0}'.format(self.spec['mpi'].mpifc),
                 'FL = {0}'.format(self.spec['mpi'].mpifc),
                 "SCALAP = %s" % scalapack.ld_flags,
                 "MUMPS_TYPE = par"])
        else:
            makefile_conf.extend(
                ["CC = cc",
                 "FC = fc",
                 "FL = fc",
                 "MUMPS_TYPE = seq"])
```

I guess that "fc" is here referring to the SPACK compiler wrapper fc and not to the Linux command "/usr/bin/fc". What is a bit puzzling regarding the error message ("/bin/sh: line 0: fc: -h: invalid option") is, that I can not figure out where the -h option is set for the fc command at all. I did a grep on all files in the staged directory where SPACK was supposed to compile the library.

Could any other SPACK user yet successfully install serial mumps ?

```
[spackapps@eu-c7-042-05 scripts]$ spack spec --install-status mumps@5.1.1+scotch~ptscotch~mpi+metis~parmetis+float+double+complex ^scotch@6.0.3+esmumps~mpi %$SSTDCOMP arch=$OS_ARCH
Input spec
--------------------------------
     mumps@5.1.1+complex+double+float+metis~mpi~parmetis~ptscotch+scotch
         ^scotch@6.0.3%gcc@4.8.5+esmumps~mpi arch=linux-centos7-x86_64 

Concretized
--------------------------------
     mumps@5.1.1%gcc@4.8.5 cflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cppflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cxxflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" +complex+double+float~int64+metis~mpi~parmetis~ptscotch+scotch+shared arch=linux-centos7-x86_64 
[+]      ^metis@5.1.0%gcc@4.8.5 cflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cppflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cxxflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2"  build_type=Release ~gdb~int64 patches=4991da938c1d3a1d3dea78e49bbebecba00273f98df2a656e38b83d55b281da1 ~real64+shared arch=linux-centos7-x86_64 
[+]          ^cmake@3.11.1%gcc@4.8.5 cflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cppflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cxxflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" ~doc+ncurses+openssl+ownlibs~qt arch=linux-centos7-x86_64 
[+]              ^ncurses@6.0%gcc@4.8.5 cflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cppflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cxxflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2"  patches=f84b2708a42777aadcc7f502a261afe10ca5646a51c1ef8b5e60d2070d926b57 ~symlinks~termlib arch=linux-centos7-x86_64 
[+]                  ^pkgconf@1.4.0%gcc@4.8.5 cflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cppflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cxxflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2"  arch=linux-centos7-x86_64 
[+]              ^openssl@1.0.1e%gcc@4.8.5 cflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cppflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cxxflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" +systemcerts arch=linux-centos7-x86_64 
[+]      ^openblas@0.2.20%gcc@4.8.5 cflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cppflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cxxflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2"  cpu_target= ~ilp64 patches=47cfa7a952ac7b2e4632c73ae199d69fb54490627b66a62c681e21019c4ddc9d +pic+shared threads=none ~virtual_machine arch=linux-centos7-x86_64 
[+]      ^scotch@6.0.3%gcc@4.8.5 cflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cppflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cxxflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" +compression+esmumps~int64+metis~mpi+shared arch=linux-centos7-x86_64 
[+]          ^bison@3.0.4%gcc@4.8.5 cflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cppflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cxxflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2"  patches=b72914fe38e54a6fc25f29019e0a0786705c4f61ce20d414cc2010c8d62448c7 arch=linux-centos7-x86_64 
[+]              ^m4@1.4.18%gcc@4.8.5 cflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cppflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cxxflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2"  patches=3877ab548f88597ab2327a2230ee048d2d07ace1062efe81fc92e91b7f39cd00 +sigsegv arch=linux-centos7-x86_64 
[+]                  ^libsigsegv@2.11%gcc@4.8.5 cflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cppflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cxxflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2"  arch=linux-centos7-x86_64 
[+]          ^flex@2.6.4%gcc@4.8.5 cflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cppflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cxxflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" +lex arch=linux-centos7-x86_64 
[+]              ^gettext@0.19.8.1%gcc@4.8.5 cflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cppflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cxxflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" +bzip2+curses+git~libunistring+libxml2 patches=9acdb4e73f67c241b5ef32505c9ddf7cf6884ca8ea661692f21dca28483b04b8 +tar+xz arch=linux-centos7-x86_64 
[+]                  ^bzip2@1.0.6%gcc@4.8.5 cflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cppflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cxxflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" +shared arch=linux-centos7-x86_64 
[+]                  ^libxml2@2.9.4%gcc@4.8.5 cflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cppflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cxxflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" ~python arch=linux-centos7-x86_64 
[+]                      ^xz@5.2.3%gcc@4.8.5 cflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cppflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cxxflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2"  arch=linux-centos7-x86_64 
[+]                      ^zlib@1.2.11%gcc@4.8.5 cflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cppflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cxxflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" +optimize+pic+shared arch=linux-centos7-x86_64 
[+]                  ^tar@1.29%gcc@4.8.5 cflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cppflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cxxflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2"  patches=08921fcbd732050c74ddf1de7d8ad95ffdbc09f8b4342456fa2f6a0dd02a957c,5c314db58d005043bf407abaf25eb9823b9032a22fd12a0b142d4bf548130fa4,d428578be7fb99b831eb61e53b8d88a859afe08b479a21238180899707d79ce4,e34d2470461037933737c0a60fbcc5d5ea905fefb6b414b4a2e88a440fcae23a arch=linux-centos7-x86_64 
[+]              ^help2man@1.47.4%gcc@4.8.5 cflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cppflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2" cxxflags="-ftree-vectorize -O2 -march=core-avx2 -fPIC -mavx2"  arch=linux-centos7-x86_64 

[spackapps@eu-c7-042-05 scripts]$ 
```

Any help is appreciated.

Best regards

Sam

**Comment by becker33** [comment]
@samfux84 can you try replacing the lines `"FC = fc"` and `"FL = fc"` with `"FC = %s" % env['FC']` and `"FL = %s" % env['FC']`, respectively?

**Comment by mcneish1** [comment]
This issue was fixed for me by #13497

**Comment by samfux84** [comment]
I can confirm #13497 fixes the issue
